File: EMS/Network/EMSSession.m
//
//  EMSSession.m
//

#import "EMSSession.h"

@implementation EMSSession

- (NSString *)description {
    NSMutableString *description = [NSMutableString stringWithFormat:@"<%@: ", NSStringFromClass([self class])];
    [description appendFormat:@"self.href=%@", self.href];
    [description appendFormat:@", self.format=%@", self.format];
    [description appendFormat:@", self.body=%@", self.body];
    [description appendFormat:@", self.state=%@", self.state];
    [description appendFormat:@", self.audience=%@", self.audience];
    [description appendFormat:@", self.keywords=%@", self.keywords];
    [description appendFormat:@", self.title=%@", self.title];
    [description appendFormat:@", self.language=%@", self.language];
    [description appendFormat:@", self.summary=%@", self.summary];
    [description appendFormat:@", self.level=%@", self.level];
    [description appendFormat:@", self.videoLink=%@", self.videoLink];
    [description appendFormat:@", self.speakers=%@", self.speakers];
    [description appendFormat:@", self.attachmentCollection=%@", self.attachmentCollection];
    [description appendFormat:@", self.speakerCollection=%@", self.speakerCollection];
    [description appendFormat:@", self.roomItem=%@", self.roomItem];
    [description appendFormat:@", self.slotItem=%@", self.slotItem];
    [description appendString:@">"];
    return description;
}

@end

File: EMS/Network/Model/EMSSession.swift
import Foundation

@objc class EMSSession: NSObject {
    var href: NSURL?

    var format: String?
    var body: String?
    var state: String?
    var audience: String?
    var keywords: [String]?
    var title: String?
    var language: String?
    var summary: String?
    var level: String?

    var link: NSURL?
    var videoLink: NSURL?

    var speakers: [EMSSpeaker]?

    var attachmentCollection: NSURL?
    var speakerCollection: NSURL?

    var roomItem: NSURL?
    var slotItem: NSURL?


    override var description: String {
        return "<\(self.dynamicType): self.href=\(self.href), self.format=\(self.format), self.body=\(self.body), self.state=\(self.state), self.audience=\(self.audience), self.keywords=\(self.keywords), self.title=\(self.title), self.language=\(self.language), self.summary=\(self.summary), self.level=\(self.level), self.link=\(self.link), self.videoLink=\(self.videoLink), self.speakers=\(self.speakers), self.attachmentCollection=\(self.attachmentCollection), self.speakerCollection=\(self.speakerCollection), self.roomItem=\(self.roomItem), self.slotItem=\(self.slotItem)>"
    }
}
